MoreCyclopsUpgrades/API/MCUServiceS.cs

```diff
@@ -0,0 +1,185 @@
+﻿namespace MoreCyclopsUpgrades.API
+{
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Reflection;
+    using MoreCyclopsUpgrades.Managers;
+
+    public interface IMCURegistration
+    {
+        /// <summary>
+        /// Registers a <see cref="AuxManagerCreateEvent"/> method that creates returns a new <see cref="IAuxCyclopsManager"/> on demand.<para/>
+        /// This method will be invoked only once for each Cyclops sub in the game world.<para/>
+        /// Auxilary managers are always created first.
+        /// </summary>
+        /// <param name="createEvent">The create event.</param>
+        void AuxCyclopsManager(AuxManagerCreateEvent createEvent);
```

Comment:
As an alternative to delegates, this could have an option to request an interface with the necessary method, making it easier for people who aren't too comfortable with delegates and anonymous methods to simply pass in a class that handles the creation of these items.